pandas/core/frame.py

```diff
@@ -8269,9 +8267,9 @@ def _flex_arith_method(
 
                 new_data = self._dispatch_frame_op(other, op)
 
-        return self._construct_result(new_data)
+        return self._construct_result(new_data, other=other)
 
-    def _construct_result(self, result) -> DataFrame:
+    def _construct_result(self, result, other=None) -> DataFrame:
```

Comment:
```suggestion
    def _construct_result(self, result, other) -> DataFrame:
```
Might as well make this required